### ISSUE TITLE
Amplify v2 resetPassword throws an error while v1.3X works fine

### ISSUE THREAD

**Issue by PineapplePie** [aws-amplify/amplify-android#2179]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-android/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-android/discussions).

### Language and Async Model

Kotlin

### Amplify Categories

Authentication

### Gradle script dependencies

<details>

```groovy
// Put output below this line

 implementation "com.amplifyframework:aws-auth-cognito:2.0.0"
```

</details>

### Environment information

<details>

```
# Put output below this line

------------------------------------------------------------
Gradle 7.4.2
------------------------------------------------------------

```

</details>

### Describe the bug

Hey! I've migrated the project from v1.31.2 to v2.0.0. All configurations are the same, I've checked the migration guide and except for the small change in the similar flow - `confirmResetPassword()` I didn't find anything new (that I had to change in order to migrate correctly).
While the previous version of SDK gives the perfect result (a not authorized user that forgot a password, receives an email with a code for entering a new password) in 100% cases, v2 just returns me an error each time I'm trying to do the same thing.

Providing an existing email, I'm getting this:

`NotAuthorizedException{message=Failed since user is not authorized., cause=NotAuthorizedException(message=Client XYZ is configured for secret but secret was not received), recoverySuggestion=Check whether the given values are correct and the user is authorized to perform the operation.}
`
Any idea how to fix that?

### Code Snippet

```java
// Put your code below this line.
Amplify.Auth.resetPassword(email, {
            continuation.resume(Resource.Data(Unit))                 
        }, {
            continuation.resume(handleError(it))
        })
```

### Log output

<details>

```
// Put your logs below this line


```

</details>

### GraphQL Schema

<details>

```graphql
// Put your schema below this line


```

</details>

**Comment by banji180** [comment]
We are not able to reproduce this error. The api works as expected and confirmation code sent via email both when signed out and when signed in. Did you generate app client secret in your cognito user pool backend? If YES, were you able to successfully make other auth api calls using from amplify app?

**Comment by PineapplePie** [comment]
@banji180 hey! Thanks for the response! Unfortunately, it can be reproduced in 100% cases.
Yes, we do have an app client secret, it was generated a long time ago and always existed in the AWS configuration file and console with the "checked" checkbox. All other APIs (sign in/sign up/log in/sign out, confirmation after sign up and so on) work fine as it was before v2, I found the issue only with the reset password flow.

Here is a piece of my AWS file without real secrets (and with some annoying formatting, haha):
```

"CognitoUserPool\": {\n"
"               \"Default\": {\n"
"                 \"PoolId\": \”XXX\”,\n”
"                 \"AppClientId\": \”XXX\”,\n”
"                 \"AppClientSecret\": \”XXX\”,\n”
"                 \"Region\": \"us-east-1\"\n"
"               }\n"
"             },\n"
"             \"Auth\": {\n"
"               \"Default\": {\n"
"                 \"OAuth\": {\n"
"                   \"WebDomain\": \”XXX\”,\n”
"                   \"AppClientId\": \”XXX\",\n"
"                   \"AppClientSecret\": \”XXX\”,\n”
"                   \"SignInRedirectURI\": \"XXX://\",\n"
"                   \"SignOutRedirectURI\": \"XXX://\",\n"
"                   \"Scopes\": [\n"
"                     \"email\",\n"
"                     \"phone\",\n"
"                     \"profile\",\n"
"                     \"openid\",\n"
"                     \"aws.cognito.signin.user.admin\"\n"
"                   ]\n"
"                 }\n"
"               }\n
```

Maybe something was changed for the configuration file itself and we're missing it? Because we're also facing the similar issues with the app client secret on iOS after the migration to v2 (haven't created a report yet).
Except for the configuration file, I didn't find any place where I can push the app client secret to the SDK.

Any thoughts on this?

**Comment by banji180** [comment]
@PineapplePie it seems that users created via the app works as expected while the ones created via the cognito backend is the one giving the Unauthorized exception. Can you please confirm whether you create the user using the cognito app?

Btw, happy new year!

**Comment by PineapplePie** [comment]
@banji180 happy new year! 🎆 

I confirm that we're creating a new user with our Cognito backend. App is responsible only for sending sign up/sign requests with provided credentials. What should we do next then? 👀

**Comment by banji180** [comment]
Hi there,
We found the issue and now working on the fix for our upcoming release. I will keep you posted once we push it.

**Comment by banji180** [comment]
Thank you for your patience, the fix has been released in v2.2.0.
